cmd/skaffold/app/cmd/cmd.go

```diff
@@ -268,8 +268,10 @@ func setUpLogs(stdErr io.Writer, level string, timestamp bool) error {
 
 func alwaysSucceedWhenCancelled(ctx context.Context, cfg sErrors.Config, err error) error {
 	// if the context was cancelled act as if all is well
-	if err == nil || ctx.Err() == context.Canceled {
+	if err != nil && ctx.Err() == context.Canceled {
 		return nil
+	} else if err == nil || err == context.Canceled {
+		return err
```

Comment:
Is this right?  If `err == context.Canceled` then we'll return `context.Canceled`.

Should this be:
```
  if err == nil || ctx.Err() == context.Canceled || err = context.Canceled {
    return nil
  }
  return sErrors.ShowAIError(cfg, err)
```

Reply:
> I'm missing the "Cluster Internal Service Error code"

This is really strange. 
```
➜  skaffold git:(fix_helm_2) git push d fix_helm_2 
Everything up-to-date
➜  skaffold git:(fix_helm_2) git push d fix_helm_2
```
And i see them locally. let me rebase on #5537 and try again.

Reply:
> Is this right? If `err == context.Canceled` then we'll return `context.Canceled`.
> 
> Should this be:
> 
> ```
>   if err == nil || ctx.Err() == context.Canceled || err = context.Canceled {
>     return nil
>   }
>   return sErrors.ShowAIError(cfg, err)
> ```

I am not sure about this condition. On master right now [here](https://github.com/GoogleContainerTools/skaffold/blob/f241521cbcdea86fc93d5f5a9528e048f529d86e/cmd/skaffold/app/cmd/cmd.go#L272) the `alwaysSucceedWhenCancelled` method returns `context.Cancelled` error. 
The test relies on this behavior [here](https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/dev_test.go#L109)

I actually went and did you mentioned and suddenly all tests started failing hence i decided to not change it. 
I think, the main, ignore cancelled error below. 
https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/skaffold.go#L38

Reply:
That test is checking the result of `doDev()`, which should return `context.Canceled`.

That code in skaffold's main is funny as we should be passing in our context.

My real point here is that `alwaysSucceedWhenCancelled` doesn't always succeed when cancelled :-)

Reply:
yep added a TODO to fix this. 
#5686